File: 19-05-2024_Find_the_maximum_sum_of_node_values.java
class Solution {
    public long maximumValueSum(int[] nums, int k, int[][] edges) {
        long sum = 0;
        long cnt = 0;
        long temp = Long.MAX_VALUE;

        for (int it : nums) {
            int a = it ^ k;
            long b = (long) a - it;
            if (a > it) {
                sum += a;
                temp = Math.min(temp, b);
                cnt++;
            } else {
                sum += it;
                temp = Math.min(temp, -b);
            }
        }

        if (cnt % 2 == 0) return sum;

        return sum - temp;
    }
}

File: 19-05-2024_Find_the_maximum_sum_of_node_values.py
class Solution(object):
    def maximumValueSum(self, nums, k, edges):
    
        sum = 0
        cnt = 0
        temp = float('inf')

        for it in nums:
            a = it ^ k
            b = a - it
            if a > it:
                sum += a
                temp = min(temp, b)
                cnt += 1
            else:
                sum += it
                temp = min(temp, -b)

        if cnt % 2 == 0:
            return sum

        return sum - temp
